ZPO1.2/Pesel.java

```diff
@@ -0,0 +1,60 @@
+public class Pesel {
+
+    private String birthDate;
+    private String gender;
+    private String pesel;
+
+    public Pesel(String newPesel){
+        try{
+            this.pesel = newPesel;
+            if(Pesel.validate(pesel)){
+                //birth date
+                if((pesel.charAt(2) == '0') || (pesel.charAt(2) == '1')){
+                    this.birthDate = String.format("%s%s.%s%s.19%s%s", pesel.charAt(4), pesel.charAt(5), pesel.charAt(2), pesel.charAt(3), pesel.charAt(0), pesel.charAt(1));
+                }else if((pesel.charAt(2) == '8') || (pesel.charAt(2) == '9')){
+                    this.birthDate = String.format("%s%s.%s%s.18%s%s", pesel.charAt(4), pesel.charAt(5), pesel.charAt(2) - '8', pesel.charAt(3), pesel.charAt(0), pesel.charAt(1));
+                }else if((pesel.charAt(2) == '2') || (pesel.charAt(2) == '3')){
+                    this.birthDate = String.format("%s%s.%s%s.20%s%s", pesel.charAt(4), pesel.charAt(5), pesel.charAt(2) - '2', pesel.charAt(3), pesel.charAt(0), pesel.charAt(1));
+                }else if((pesel.charAt(2) == '4') || (pesel.charAt(2) == '5')){
+                    this.birthDate = String.format("%s%s.%s%s.21%s%s", pesel.charAt(4), pesel.charAt(5), pesel.charAt(2) - '4', pesel.charAt(3), pesel.charAt(0), pesel.charAt(1));
+                }else if((pesel.charAt(2) == '6') || (pesel.charAt(2) == '7')){
+                    this.birthDate = String.format("%s%s.%s%s.22%s%s", pesel.charAt(4), pesel.charAt(5), pesel.charAt(2) - '6', pesel.charAt(3), pesel.charAt(0), pesel.charAt(1));
+                }
+                //gender
+                if(((int) pesel.charAt(9) % 2) == 0){
+                    this.gender = "female";
+                }else{
+                    this.gender = "male";
+                }
+            }else{
+                throw new IllegalArgumentException();
+            }
+        }catch (Exception e){
+            System.out.println("Incorrect pesel number\nProgramme will will shut down");

```

Comment:
@repcus Zamiast łapać tutaj ten wyjątek to można udostępnić go na zewnątrz i wtedy inny programista, który dostanie ten kod może podjąć samemu decyzję co ma z tym zrobić.